### PR TITLE
[ticket/16544] Use forum_notify.txt for forum subscription email notifications

### PR DESCRIPTION
**Pull request by rxu**
Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

<a href="https://tracker.phpbb.com/browse/PHPBB3-16544">PHPBB3-16544</a>.

### CODE REVIEW

**Comment by rxu** [comment]
I'm not sure if migration is needed here.

File: phpBB/phpbb/notification/type/forum.php [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	*/
+	public function get_type()
+	{
+		return 'notification.type.forum';
+	}
+
+	/**
+	* Notification option data (for outputting to the user)
+	*
+	* @var bool|array False if the service should use it's default data
```
Comment (marc1706) [inline phpBB/phpbb/notification/type/forum.php:36]:
```suggestion
	* @var bool|array False if the service should use its default data
```

File: phpBB/phpbb/notification/type/forum.php [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		$this->db->sql_freeresult($result);
+
+		$notify_users = $this->get_authorised_recipients($users, $post['forum_id'], $options, true);
+
+		if (empty($notify_users))
+		{
+			return [];
+		}
+
+		// Try to find the users who already have been notified about replies and have not read the topic since and just update their notifications
```
Comment (marc1706) [inline phpBB/phpbb/notification/type/forum.php:79]: Topic? :D

Comment (rxu) [reply phpBB/phpbb/notification/type/forum.php:79]: Actually yes, as notification takes the user to unread posts in the topic.

**Pull request merged**